tests/test_repo.py

```diff
@@ -3,164 +3,53 @@
-            assert result.returncode == 0
-            assert "packages" in result.stdout
-
-    def test_pyproject_versions(
-        self,
-    ):
-        """Test the pyproject.toml versions are updated."""
-
-        # We read in the pyproject.toml file and check the versions
-        current_pyproject = self.parent_dir / "pyproject.toml"
-        repo_pyproject = (
-            self.parent_dir / "auto_dev" / "data" / "repo" / "templates" / "autonomy" / "pyproject.toml.template"
-        )  # pylint: disable=line-too-long
-
-        auto_dev_deps = toml.loads(current_pyproject.read_text())["tool"]["poetry"]["dependencies"]
-        repo_deps = toml.loads(
-            repo_pyproject.read_text().format(
-                project_name=self.repo_name,
-                author=self.author,
-            )
-        )["tool"]["poetry"]["dependencies"]
-
-        errors = []
-        for key in auto_dev_deps:
-            if key not in repo_deps:
-                continue
-            if auto_dev_deps[key] in repo_deps[key]:
-                continue
-            val = f"{key} New: {auto_dev_deps[key]} old: {repo_deps[key]}"
-            errors.append(val)
-        assert not errors, errors
+def _test_repo_scaffold(repo_type, make_commands, cli_runner, test_clean_filesystem):
+    repo_root = Path(test_clean_filesystem) / "dummy"
+    command = ["adev", "repo", "scaffold", repo_root.name, "-t", repo_type]
+    runner = cli_runner(command)
+    result = runner.execute()
+    makefile = repo_root / "Makefile"
+
+    # Verify the basic repository structure exists
+    assert result, runner.output
+    assert repo_root.exists(), f"Repository {repo_root} does not exist."
+    assert (repo_root / ".git").exists(), f".git directory not found in {repo_root}."
+    assert makefile.exists(), f"Makefile not found in {repo_root}."
+
+    # Run each make command and collect any errors.
+    error_messages = {}
+    for cmd in make_commands:
+        proc_result = subprocess.run(
+            ["make", cmd],
+            shell=False,
+            capture_output=True,
+            text=True,
+            check=False,
+            cwd=repo_root,
+        )
+        if proc_result.returncode != 0:
+            error_messages[cmd] = proc_result.stderr
+    assert not error_messages, f"Errors encountered in make commands: {error_messages}"
+
+
+def test_python_repo(cli_runner, test_clean_filesystem):
+    """Test scaffolding a Python repository."""
+
+    _test_repo_scaffold(
+        repo_type="python",
+        make_commands=("fmt", "lint", "test"),
+        cli_runner=cli_runner,
+        test_clean_filesystem=test_clean_filesystem,
+    )
+
+
+def test_autonomy_repo(cli_runner, test_clean_filesystem):
+    """Test scaffolding an Autonomy repository."""
+
+    _test_repo_scaffold(
+        repo_type="autonomy",
+        make_commands=("fmt", "lint", "test", "hashes"),
+        cli_runner=cli_runner,
+        test_clean_filesystem=test_clean_filesystem,
+    )
```

Comment:
Rewrote the repo scaffolding tests to eliminate duplication, flakiness, and `pytest-xdist` interference. Each repo type now has a single test that covers end-to-end scaffolding and Makefile validation. Dropped the “fail if repo exists” test (not worth the cost of invoking `isolated_filesystem(copy=True)`) and the `pyproject.toml` version comparison (no longer needed since templates are generated dynamically)